docs/AndroidBuildingFromSource.md

```diff
@@ -148,4 +148,6 @@ If you made changes to React Native and submit a pull request, all tests will ru
 
 ## Troubleshooting
 
-Gradle build fails in `ndk-build`. See the section about `local.properties` file above.
+- Gradle build fails in `ndk-build`. See the section about `local.properties` file above.
+
+- Wrong react-native version is used (0.20): if you're using dependencies that have a reference to react-native, you need to run the following command first: `./gradlew :ReactAndroid:installArchives`. This will build react-native in node_modules, so that it can be properly referenced from your dependencies. If such a module does not find anything at node_modules/react-native, it will probably search the Maven repository, where it will find react-native verion 0.20, resulting in a lot of errors.
```

Comment:
`compile 'com.facebook.react:react-native:+'` is indeed the better way to reference dependencies, but the problem we faced was the following:

In f.e. react-native-maps, the repositories are defined like this:

`repositories {
      mavenLocal()
      jcenter()
      maven {
          // For developing the library outside the context of the example app, expect `react-native`
          // to be installed at `./node_modules`.
          url "$projectDir/../node_modules/react-native/android"
      }
      maven {
          // For developing the example app.
          url "$projectDir/../../react-native/android"
      }
  }`

What happens here, is that gradle will search in all these repositories for react-native, but since it can't find anything in node_modules/react-native/android (no build has happened yet), it will take the version it CAN find, which is the one on Maven. When you're building Android from source, you have to run this command first to build react-native, resulting in an /android folder: `./gradlew :ReactAndroid:installArchives`.

Just thought it would be nice to put it here, since we spent quite some time searching for this :)